packages/@n8n/nodes-langchain/nodes/agents/Agent/agents/ToolsAgent/execute.ts

```diff
@@ -264,7 +264,7 @@ export async function toolsAgentExecute(this: IExecuteFunctions): Promise<INodeE
 	const hasBinaryData = this.getInputData(0, 'main')?.[0]?.binary !== undefined;
 	if (hasBinaryData && passthroughBinaryImages) {
 		const binaryMessage = await extractBinaryMessages(this);
-		messages.push(binaryMessage);
+		messages.splice(3, 0, binaryMessage);
```

Comment:
Could you please add a comment explaining why we're inserting it after 3rd item specifically? Just to make sure this doesn't break if we add something into `messages`

Reply:
I added the scratchpad after the binary message instead, and left a comment there too. Thanks for your review!